compiler/noirc_frontend/src/tests.rs

```diff
@@ -4073,3 +4073,25 @@ fn regression_7088() {
     "#;
     assert_no_errors(src);
 }
+
+#[test]
+fn errors_if_loop_does_not_have_break() {
+    let src = r#"
+    fn main() {
+        /// Safety: test
+        unsafe {
+            foo()
+        }
+    }
+
+    unconstrained fn foo() {
+        loop {}
```

Comment:
Please add 1-2 non-trivial cases, e.g.
```noir
loop {
  x += 1;
  bar(x);
}
```
or a case where there are 2+ nested loops where only only nesting level is missing a `break`, to test the `self.nested_loops`/`self.current_loop` logic.

Reply:
Done!